namesys/validator.go

```diff
@@ -94,6 +96,34 @@ func (v IpnsValidator) Validate(key string, value []byte) error {
 	return nil
 }
 
+func (v IpnsValidator) getPublicKey(pid peer.ID, entry *pb.IpnsEntry) (ic.PubKey, error) {
+	if entry.PubKey != nil {
+		pk, err := ic.UnmarshalPublicKey(entry.PubKey)
+		if err != nil {
+			// TODO: i think this counts as a 'malformed record' and should be discarded
+			log.Debugf("public key in ipns record failed to parse: ", err)
+			return nil, err
+		}
+		expPid, err := peer.IDFromPublicKey(pk)
```

Comment:
This could fail if the user manually created the peerID differently than this function does (for example, made their key the hash of a public key that this function would choose to embed). I'm not sure how much of a problem is